src/redux/books/books.js

```diff
@@ -0,0 +1,26 @@
+// ACTIONS
+const ADD = 'my-app/widgets/ADD';
+const REMOVE = 'my-app/widgets/REMOVE';
+
+export default function booksReducer(state = {}, action = {}) {
+  switch (action.type) {
+    case ADD:
+      return [
+        ...state,
+        {
+          id: action.id,
+          book: action.book,
+          author: action.author,
+        },
+      ];
+    case REMOVE:
+      return state.filter((book) => book.id !== action.id);
+
+    default:
+      return state;
+  }
+}
+
+export const addWidgets = (widget) => ({ type: ADD, widget });
+
+export const removeWidgets = (widget) => ({ type: REMOVE, widget });
```

Comment:
- [ ] **[OPTIONAL]** I think it would be more appropriate to rename these action creators, `addWidgets` and `removeWidgets`, to `addBook` and `removeBook` respectively.

- [ ] For the **addWidgets** action between lines 10 and 14 you are calling `action.id` and so on which you have not defined. What you have is only `widget`, so we are likely to encounter errors when adding a book. For you implementation to work in the **addWidgets** function you need to define `id`, `book` and `author`, instead of only `widget`:

  ```
  export const addBook = (payload) => ({
    type: ADD,
    id: payload.id,
    title: payload.title,
    author: payload.author
  })
  ```
  Better still we can simply have this:
  ```
  export const addBook = (payload) => ({ type: ADD, payload})
  ```
  where the payload is the book object, then we change lines 7 to 15 to this.
  ```
  case ADD:
    return [
      action.payload,
      ...state
    ];
  ```

- [ ] Kindly take the same approach for the **removeWidget** method and ensure the `action.id` you are calling on line 17 is actually defined.

  ```
  export const removeBook = (id) => ({ type: REMOVE, id })
  ```